packages/compass-connections/src/stores/connections-store.tsx

```diff
@@ -211,13 +236,53 @@ export function useConnections({
   );
   const { activeConnectionId } = state;
 
+  const getActiveAndInactiveCount = useCallback(() => {
+    const total = favoriteConnections.length + recentConnections.length;
+    const active =
+      connectionsManager.getConnectionIdsByStatus(ConnectionStatus.Connected)
+        .length || 0;
+    return { active, inactive: total - active };
+  }, [favoriteConnections, recentConnections, connectionsManager]);
+
+  const findConnectionInfo = useCallback(
+    (connectionId: ConnectionInfo['id']) => {
+      return [...favoriteConnections, ...recentConnections].find(
+        ({ id }) => id === connectionId
+      );
+    },
+    [favoriteConnections, recentConnections]
+  );
+
   const saveConnectionInfo = useCallback(
-    async (
-      connectionInfo: RecursivePartial<ConnectionInfo> &
-        Pick<ConnectionInfo, 'id'>
-    ) => {
+    async ({
+      fullConnectionInfo,
+      partialConnectionInfo,
+    }:
+      | {
+          fullConnectionInfo?: undefined;
+          partialConnectionInfo: RecursivePartial<ConnectionInfo> &
+            Pick<ConnectionInfo, 'id'>;
+        }
+      | {
+          fullConnectionInfo: ConnectionInfo;
+          partialConnectionInfo?: undefined;
+        }) => {
+      const connectionInfo = fullConnectionInfo || partialConnectionInfo;
       try {
-        await saveConnection(connectionInfo);
+        if (fullConnectionInfo) {
+          const isNewConnection = !findConnectionInfo(fullConnectionInfo.id);
+          if (isNewConnection) {
+            const { active, inactive } = getActiveAndInactiveCount();
+            const isActive =
+              connectionsManager.statusOf(fullConnectionInfo.id) ===
+              'connected'; // save normally happens first, but in case.
+            onConnectionCreated?.(fullConnectionInfo, {
+              active: isActive ? active + 1 : active,
```

Comment:
This is creating an implicit hidden coupling, personally i don't like particularly this style because it corrupts code over time in ways that aren't obvious.

When we add `active` and `inactive` (btw please let's at least call them "activeConnectionsCount"), we are admitting that this code knows what `onConnectionCreated` will do. That is the same than calling `track` here, in fact i would rather prefer that.

The 2 other alternatives are instead very clear:
1. we don't change `saveConnectionInfo`, which is a critical function, that is better if it stays simple and if it doesn't fail or gets more complex for unrelated reason. And instead we could emit an `onConnectionSaved` and have the entire telemetry related logic there (including counting active connections etc ..).

2. we very explicitly call track here, which would make deadly clear why all of this computation are needed, and would make it also easier to refactor code in a simpler way, just extracting it to functions  with a clear name.